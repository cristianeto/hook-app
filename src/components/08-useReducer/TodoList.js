import React from "react";
import TodoListItem from "./TodoListItem";

const TodoList = ({ todos, onDelete, onDone }) => {
  return (
    <ul className="list-group list-group-flush">
      {todos.map((todo, index) => (
        <TodoListItem
          index={index}
          key={todo.id}
          onDelete={onDelete}
          onDone={onDone}
          todo={todo}
        />
      ))}
    </ul>
  );
};

export default TodoList;
