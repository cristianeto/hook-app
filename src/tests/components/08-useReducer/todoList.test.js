import TodoList from "../../../components/08-useReducer/TodoList";
import { shallow } from "enzyme";
import { demoTodos } from "../../fixtures/demoTodos";

describe("Testing in <TodoList/>", () => {
  const handleDelete = jest.fn();
  const handleDone = jest.fn();
  const wrapper = shallow(
    <TodoList todos={demoTodos} onDelete={handleDelete} onDone={handleDone} />
  );

  test("should return the <TodoList/> componente correctly", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("should show 2 item of <TodoListItem/>", () => {
    //console.log(wrapper.find("TodoListItem").length);
    //console.log(wrapper.find("TodoListItem").at(0).props());//? encuentra el TodoListItem de la position y muestra todas las props

    expect(wrapper.find("TodoListItem").length).toBe(demoTodos.length);

    expect(wrapper.find("TodoListItem").at(0).prop("onDelete")).toEqual(
      expect.any(Function) //?espera que sea una funcion
    );
  });
});
