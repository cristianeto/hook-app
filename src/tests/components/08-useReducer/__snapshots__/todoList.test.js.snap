// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing in <TodoList/> should return the <TodoList/> componente correctly 1`] = `
<ul
  className="list-group list-group-flush"
>
  <TodoListItem
    index={0}
    key="1"
    onDelete={[MockFunction]}
    onDone={[MockFunction]}
    todo={
      Object {
        "desc": "Aprender React",
        "done": false,
        "id": 1,
      }
    }
  />
  <TodoListItem
    index={1}
    key="2"
    onDelete={[MockFunction]}
    onDone={[MockFunction]}
    todo={
      Object {
        "desc": "Aprender Firebase",
        "done": false,
        "id": 2,
      }
    }
  />
</ul>
`;
